pipenv/cli.py

```diff
@@ -1776,7 +1776,11 @@ def install(
     # Capture -e argument and assign it to following package_name.
     more_packages = list(more_packages)
     if package_name == '-e':
-        package_name = ' '.join([package_name, more_packages.pop(0)])
+        package_name = ' '.join([package_name, more_packages.po(0)])
+
+    # Capture . argument and asign it to nothing
+    if package_name == '.':
+        package_name = False
```

Comment:
I'm pretty sure we're going to blow up farther down the line if the user passes more than one package name because we now how a list of booleans and strings.

We may want to take another approach on this or make a more stringent check that has its own logic and exit strategy.

Reply:
@nateprewitt Agree.  It was brought to my attention that the "." is commonly used on unix to specify current directory so I thought I'd add in a check for it.  It is a bit hairy now.

Reply:
installing . should be totally valid, it's specified in Pipenvs pipfile -- how will this interact with `-e`?  Also if you are in a directory that is a valid python package you technically should be allowed to `install .`.  Pip allows you to `pip install .` or `pip install -e .` I believe

Reply:
@techalchemy you're right, but I think the issue here is the user is trying to install a package with a Pipfile but no setup.py. It's the equivalent of 'pipenv install' but we currently fail on it because the users package doesn't have the machinery pip-tools expects.

I think the solution here is to try to install because the users directory could very well be a valid package but when we hit he exception warn that they aren't in an installable directory and possibly redirect them to 'pipenv install' because that's what they likely want.

That or we alias this to pipenv install when a setup.py isn't present but that's likely going to be another parsing rats nest.

Reply:
So this discussion is highly relevant because of #936 and #939 + ongoing bugs with local installations, and my proposed approach is in line with yours (slightly different context but generally the same) -- if a user passes a path that points to a _directory_ on the filesystem, it would be most idiomatic to attempt to install it (since it might be a valid package) and then fail -- in this case with a warning about not being installable, but in the case I'm thinking of we would want to check pypi to see if there is a matching package there as well (not applicable to installing '.' obviously)

Reply:
@techalchemy yeah, I think we're on the same page. We just need to get the cascade right if we're going to try different sources (local, url, pypi; not necessarily in that order) and implement it in one place.